### PR TITLE
CORE-5981: Add logs for bad SNI

### PR DESCRIPTION
**Pull request by yift-r3**

### CODE REVIEW

File: components/gateway/src/main/kotlin/net/corda/p2p/gateway/messaging/http/BaseHttpChannelHandler.kt [hunk 1]
```diff
@@ -70,6 +77,18 @@ abstract class BaseHttpChannelHandler(private val eventListener: HttpConnectionL
                         cause is SSLException && (cause.message?.contains("internal_error") == true) -> {
                             logger.warn("Received internal_error during handshake")
                         }
+                        cause is SSLException && (cause.message?.contains("unrecognized_name") == true) -> {
```
Comment (williamvigorr3) [inline components/gateway/src/main/kotlin/net/corda/p2p/gateway/messaging/http/BaseHttpChannelHandler.kt:80]: Should we merge `cause.message?.contains("unrecognized_name")` and `cause.message?.contains("Unrecognized server name indication") == true)`, as we print the same warning in both cases.

**Review by williamvigorr3 — COMMENTED** [review]
LGTM (left a very minor comment). Did you manage to (manually if necessary) test this? I am no means a netty expert, looking at the logs in the original ticket it looks like you have taken the correct approach.

File: components/gateway/src/main/kotlin/net/corda/p2p/gateway/messaging/http/BaseHttpChannelHandler.kt (hunk 1 above)
Comment (yift-r3) [reply components/gateway/src/main/kotlin/net/corda/p2p/gateway/messaging/http/BaseHttpChannelHandler.kt:80]: One happened on the server side and one on the client side, so I wanted to allow our self to change the messages...

Comment (williamvigorr3) [reply components/gateway/src/main/kotlin/net/corda/p2p/gateway/messaging/http/BaseHttpChannelHandler.kt:80]: Fair enough.

**Review by williamvigorr3 — APPROVED** [review]
LGTM

**Pull request merged**